src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1418,3 +1478,39 @@ def try_create_application_insights(cmd, resource_group, name, location):
                    'Application Insights component', appinsights.name, portal_url, appinsights.id)
 
     return appinsights.instrumentation_key
+
+
+def app_insights_update(cmd, client, resource_group, name, app_insights_key=None, app_insights=None, sampling_rate=None, disable=None, no_wait=False):
+    if disable:
+        trace_properties = models_20201101preview.MonitoringSettingProperties(trace_enabled=False)
+    else:
+        trace_properties = client.monitoring_settings.get(resource_group, name).properties
+        if not trace_properties.app_insights_instrumentation_key and not app_insights_key and not app_insights and sampling_rate:
+            CLIError("Can't set '--sampling-rate' without connecting to Application Insights. Please provide '--app-insights' or '--app-insights-key'.")
+        if app_insights_key:
+            instrumentation_key = app_insights_key
+        elif app_insights:
+            if is_valid_resource_id(app_insights):
+                resource_id_dict = parse_resource_id(app_insights)
+                instrumentation_key = get_app_insights_key(cmd.cli_ctx, resource_id_dict['resource_group'],
+                                                           resource_id_dict['resource_name'])
+            else:
+                instrumentation_key = get_app_insights_key(cmd.cli_ctx, resource_group, app_insights)
+        else:
+            instrumentation_key = trace_properties.app_insights_instrumentation_key
+        if sampling_rate:
+            trace_properties = models_20201101preview.MonitoringSettingProperties(
+                trace_enabled=True, app_insights_instrumentation_key=instrumentation_key, app_insights_sampling_rate=sampling_rate)
+        elif trace_properties.app_insights_sampling_rate:
+            trace_properties = models_20201101preview.MonitoringSettingProperties(
+                trace_enabled=True, app_insights_instrumentation_key=instrumentation_key, app_insights_sampling_rate=trace_properties.app_insights_sampling_rate)
+    if trace_properties is not None:
+        sdk_no_wait(no_wait, client.monitoring_settings.update_put,
+                    resource_group_name=resource_group, service_name=name, properties=trace_properties)
+
+
+def app_insights_show(cmd, client, resource_group, name, no_wait=False):
+    trace_properties = client.monitoring_settings.get(resource_group, name).properties
+    if not trace_properties:
+        raise CLIError("Application Insights not set.")
```

Comment:
Could it be the case that application insights is set, but user provides the wrong name?

Reply:
user can't set application insights with the wrong name.

Reply:
We check the name in 1493 line: is_valid_resource_id(app_insights)